### ISSUE TITLE
`unused-parameter` rule should not trigger on binding closure parameter

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#5740]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

The rule triggers on `$historyItem`, but the parameter is kinda used. `$historyItem` results in a binding, but it also makes it possible to access the normal variable with `historyItem` implicitly. So even though `$historyItem` is not directly used, it cannot be removed.

```swift
List($history) { $historyItem in
	Foo(
		url: historyItem.url
	)
}
```

In short, make it not trigger on `$`-prefixed parameters in a closure.

### Environment

* SwiftLint version: 0.56.0
* Xcode version: Xcode 15.4, Build version 15F31d
* Installation method used: Installer
* Configuration file:

```yml
only_rules:
  - unused_parameter
```

**Comment by SimplyDanny** [comment]
Since `$_` is allowed as a parameter, the same rules used for normal parameters apply here as well. In the example, if any of `$historyItem` or `historyItem` is used in the closure block, `$historyItem` can be considered to be used. If there is no reference at all, it can be replaced with `$_` (or `$_historyItem` once that's implemented).

However, silencing the rule completely for `$`-prefixed parameters might be too cautious.

**Comment by systemblueio** [comment]
Opened #6763 to fix this: `unused_parameter` now skips `$`-prefixed closure parameters, since the projection also binds the wrapped value that gets used without the prefix.

**Comment by SimplyDanny** [comment]
Fixed by #6889.
